syntax = "proto3";

service Game {

   rpc register (Soldier) returns (ServerOutput) {}
   rpc missile_approach (Empty) returns (stream Missile) {}
   rpc status(Request) returns (Empty){}
   rpc broadcast(Request) returns (stream Response){}
   rpc unary(Request) returns (Response){}
   rpc sendMissile(Empty) returns (Empty);
   rpc update_cordinates(Update) returns (Empty);
   rpc status_all(Empty) returns(Response);
   rpc print(Empty) returns(Response);
}
message Empty {}

message Soldier {
   int32  x = 1;
   int32  y = 2;
   int32 speed=3;
}

message Update{
   bool alive=1;
   int32 x=2;
   int32 y=3;
   string message=4;
   int32 soldierID=5;
}

message Missile {
   int32  x = 1;
   int32  y = 2;
   int32 rad=3;
}
message ServerOutput {
   string message = 1;
}

message Request {
  string message = 1;
}

message Response {
  string message = 1;
}

